### ISSUE TITLE
can't import corpus from tab-delimited file

### ISSUE THREAD

**Issue by bhallen** [PhonologicalCorpusTools/CorpusTools#453]
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/corpustools-1.1.0-py3.4.egg/corpustools/gui/iogui.py", line 107, in run
    corpus = load_corpus_csv(**self.kwargs)
  File "/usr/local/lib/python3.4/dist-packages/corpustools-1.1.0-py3.4.egg/corpustools/corpus/io/csv.py", line 166, in load_corpus_csv
    corpus.add_word(word)
  File "/usr/local/lib/python3.4/dist-packages/corpustools-1.1.0-py3.4.egg/corpustools/corpus/classes/lexicon.py", line 2502, in add_word
    self.update_inventory(word.transcription)
  File "/usr/local/lib/python3.4/dist-packages/corpustools-1.1.0-py3.4.egg/corpustools/corpus/classes/lexicon.py", line 2531, in update_inventory
    if transcription.stress_pattern:
AttributeError: 'str' object has no attribute 'stress_pattern'

**Comment by bhallen** [comment]
I'm working on this now, but could someone please try to replicate the bug? I just tried to import our standard example.txt corpus. It seems odd that we'd have such a big problem while all our corpus import tests still pass...

**Comment by bhallen** [comment]
@mmcauliffe 
This is fixed now. The transcription strings weren't getting turned into Transcriptions because they weren't caught by the elif on 932 of lexicon.py. I don't understand exactly how this att.att_type setup works, but the att_type of transcription atts were all 'factor' rather than 'tier'. I allowed the script to consider 'factor' type attributes as needing to be Transcriptions.

**Comment by bhallen** [comment]
@mmcauliffe 
I've left my "fix" uncommitted because it caused other problems, like PCT thinking that imported corpora have no feature system even when one is chosen and all segments are specified by it.

Digging deeper, it seems like all this is due to the transcription column of a corpus somehow getting Factor as its AttributeType instead of Tier. But I can't comprehend the overarching logic behind what's going on with AttributeTypes and related classes without discussing it with you. So (assuming you can replicate the original bug) as the one who created the classes, could you see if you can fix this?

**Comment by mmcauliffe** [comment]
Which version of the example corpus are you using?  The CorpusTools/tests/data/csv/example.txt?  That one works just fine.  My main guess is that the corpus that you're using doesn't have delimiters, so PCT can't figure out that it's a transcription column.  I think the bug arises because the transcription attribute is unfortunately still hard coded in.

**Comment by bhallen** [comment]
I was using the example corpus I've always used (albeit almost exclusively for command line stuff), but you're right that the one in tests/data/csv seems to work without any problems. Thanks for letting me know.

Does this mean that PCT is currently unable to read in column-delimited corpora files that don't have explicit delimiters?

**Comment by mmcauliffe** [comment]
No, not at all.  You just have to explicitly tell it that it's a transcription column (and not a character column of some type, like part of speech or something) with no transcription delimiter.  It just has no way of detecting transcription vs character when there are no delimiters. I'll add some error checking code such that protected names like 'transcription' shouldn't be character columns.

**Comment by mmcauliffe** [comment]
Just pushed a commit to master that should add an error message for this.  Can you double check the original example.txt that you were using, and make sure this catches it?  Thanks!

**Comment by bhallen** [comment]
Oh, I see, so I can change "Annotation type" in the parsing preview sub-window... I never would have thought of that. It makes sense now, though. I agree about defaulting to the transcription type for a column named 'transcription', though. It might also be good to change the "Annotation type" label on that drop-down box in the parsing preview sub-window, since that won't mean anything to someone who isn't familiar with the underlying code.

I'll re-open this issue and assign myself to it. As soon as I have time after working out the other two issues I'm assigned, I'll see if I can make this process friendlier to our users.

**Comment by bhallen** [comment]
Yep, your commit caught it! Thanks again. I just pushed a little tweak to the text of the message box that directs users to where they can change the annotation type.

At this point, I think the only thing that could be improved related to this issue is to have columns with the transcription header default to the transcription annotation type, so I'll do that.

**Comment by bhallen** [comment]
also be sure users are warned if there is no transcription (no tier annotationtype) in corpus

**Comment by bhallen** [comment]
Importing a corpus using the CL now works even if the corpus has no explicit transcription delimiters (as long as the transcription column is called 'Transcription' or 'transcription'

And I tried my best to warn GUI users when they're about to import a corpus with no Transcription AttributeType columns, but after all the research into the whole importing pipeline and QtWidgets, I got stymied by how Qt handles threads. Could someone who's more gui-savvy add this little warning message in? I'm going to assign @jsmackie for now.
